evm.go

```diff
@@ -0,0 +1,139 @@
+package chain_selectors
+
+import (
+	_ "embed"
+	"fmt"
+	"strconv"
+
+	"gopkg.in/yaml.v3"
+)
+
+//go:generate go run genchains_evm.go
+
+//go:embed selectors.yml
+var selectorsYml []byte
+
+//go:embed test_selectors.yml
+var testSelectorsYml []byte
+
+type ChainDetails struct {
+	ChainSelector uint64 `yaml:"selector"`
+	ChainName     string `yaml:"name"`
+}
+
+var (
+	evmSelectorsMap           = parseYml(selectorsYml)
+	evmTestSelectorsMap       = parseYml(testSelectorsYml)
+	evmChainIdToChainSelector = loadAllEVMSelectors()
+	evmChainsBySelector       = make(map[uint64]Chain)
+	evmChainsByEvmChainID     = make(map[uint64]Chain)
+)
+
+func init() {
+	for _, ch := range ALL {
+		evmChainsBySelector[ch.Selector] = ch
+		evmChainsByEvmChainID[ch.EvmChainID] = ch
+	}
+}
+
+func loadAllEVMSelectors() map[uint64]ChainDetails {
+	output := make(map[uint64]ChainDetails, len(evmSelectorsMap)+len(evmTestSelectorsMap))
+	for k, v := range evmSelectorsMap {
+		output[k] = v
+	}
+	for k, v := range evmTestSelectorsMap {
+		output[k] = v
+	}
+	return output
+}
+
+func parseYml(ymlFile []byte) map[uint64]ChainDetails {
+	type ymlData struct {
+		SelectorsByEvmChainId map[uint64]ChainDetails `yaml:"selectors"`
+	}
+
+	var data ymlData
+	err := yaml.Unmarshal(ymlFile, &data)
+	if err != nil {
+		panic(err)
+	}
+
+	return data.SelectorsByEvmChainId
+}
+
+func EvmChainIdToChainSelector() map[uint64]uint64 {
+	copyMap := make(map[uint64]uint64, len(evmChainIdToChainSelector))
+	for k, v := range evmChainIdToChainSelector {
+		copyMap[k] = v.ChainSelector
+	}
+	return copyMap
+}
+
+func ChainIdFromSelector(chainSelectorId uint64) (uint64, error) {
```

Comment:
Recommendation: mark this function as deprecated and create a parallel func `EVMChainIDFromChainSelector` to make it clear this is EVM which should be used instead. The "chain ID" concept should now be abstract and be more scoped with the chain family that you want a chain ID for. Feels like someone can easily use this without intending to and run into issues.

```golang
func EVMChainIDFromSelector(chainSelector uint64) (*big.Int, error) {
// .. impl
// btw, chain IDs are technically uint256
}

// Deprecated: use EVMChainIDFromSelector instead
func ChainIdFromSelector(chainSelectorId uint64) (uint64, error) {
// ...
}
```

Reply:
I wonder if it might be easier conceptually to just leave it as "Chain" is an EVM chain and then otherwise have a family prefix like SolXXX?